### PR TITLE
Added ImageSequence all_frames

### LINKED ISSUE #3758: Transforms and enhancements across all frames
It would be nice if transforms and enhancements could be applied on animations.

### PR DESCRIPTION
**Pull request by radarhere**
Resolves #3758 

Adds a new method to apply a supplied function to all frames of an image at once. So, for example,

```python
from PIL import Image, ImageSequence
im = Image.open('Tests/images/g4-multi.tiff')

ims = ImageSequence.all_frames(im, lambda im_frame: im_frame.rotate(180))

ims[0].save('out.tiff', save_all=True, append_images=ims[1:])
```

### CODE REVIEW

**Comment by pirate486743186** [comment]
hmmm.... Maybe add multi-threading support?
The function could take a third parameter, the maximum number of threads to be used, default to one. 

If no time for this, then you could just silently add the third parameter anyway, so that future code is backwards compatible with the first version of the function.

**Comment by radarhere** [comment]
While we are concerned with backwards compatibility, I don't think that we are concerned with forwards compatibility.

Nothing else in the Pillow runtime uses multiprocessing, so I'm not personally inclined to integrate it here. If anyone else feels otherwise, feel free to comment. Multiprocessing can't be applied to the process of splitting up a single image into frames, because it uses the same file pointer, but it could be applied to the supplied function. If a user wanted to do that, they could just -

```python
from PIL import Image, ImageSequence
im = Image.open('Tests/images/iss634.gif')

ims = ImageSequence.all_frames(im)

from multiprocessing import Pool
def f(im_frame):
    return im_frame.rotate(180)
ims = Pool(5).map(f, ims)

ims[0].save('out.gif', save_all=True, append_images=ims[1:])
```

**Comment by pirate486743186** [comment]
but isn't this a special case? You are basically processing multiple images at once. And it seams it would be used frequently IMHO.

File: docs/releasenotes/6.1.0.rst [hunk 1]
```diff
@@ -11,6 +11,14 @@ An optional ``include_layered_windows`` parameter has been added to ``ImageGrab.
 defaulting to ``False``. If true, layered windows will be included in the resulting
 image on Windows.
 
+ImageSequence.all_frames
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+A new method to facilitate applying a given function to all frames in an image, or to
+all frames in a list of images. The frames are returned as a list of separate images.
+For example, ``ImageSequence.all_frames(im, lambda im_frame: im_frame.rotate(90))``
+could used to return all frames from an image, each rotated 90 degrees.
```
Comment (hugovk) [inline docs/releasenotes/6.1.0.rst]:
```suggestion
could be used to return all frames from an image, each rotated 90 degrees.
```

Comment (radarhere) [reply docs/releasenotes/6.1.0.rst]: Thanks

**Pull request merged**